peps/pep-0733.rst

```diff
@@ -0,0 +1,651 @@
+with different requirements and areas of focus. This section
+describes this complex state of affairs in terms of the
+actions that different stakeholders need to perform through
+the C API.
+
+Universal Actions for All Stakeholders
+--------------------------------------
+
+There are actions which are generic, and required by
+all types of API users:
+
+* Define functions and call them
+* Define new types
+* Create instances of builtin and user-defined types
+  and perform operations on them
+* Introspect objects, including types, instances, and functions
+* Raise and handle exceptions
+* Import modules
+* Manage threads
+* Manage sub-interpreters
+* Handle and send signals
+
+The following sections look at the unique requirements of various stakeholders.
+
+Extension Writers
+-----------------
+
+Extension writers are the traditional users of the C API. Their requirements
+are the universal actions listed above.
+
+Authors of Embedded Python Applications
+---------------------------------------
+
+Applications with an embedded Python interpreter. Examples are
+`Blender <https://docs.blender.org/api/current/info_overview.html>`__ and
+`OBS <https://obsproject.com/wiki/Getting-Started-With-OBS-Scripting>`__.
+
+They need to be able to:
+
+* Configure the interpreter (import paths, inittab, ``sys.argv``, etc.).
+* Interact with the execution model and program lifetime, including
+  clean interpreter shutdown and restart.
+* Represent complex data models in a way Python can use without
+  having to create deep copies.
+* Provide and import frozen modules.
+* Run multiple independent interpreters (in particular, when embedded
+  in a library that wants to avoid global effects).
+
+Alternative Python Implementations
+----------------------------------
+
+Alternative implementations of Python (such as
+`PyPy <https://www.pypy.org>`__,
+`GraalPy <https://www.graalvm.org/python/>`__,
+`IronPython <https://ironpython.net>`__,
+`RustPython <https://github.com/RustPython/RustPython>`__,
+`MicroPython <https://micropython.org>`__,
+and `Jython <https://www.jython.org>`__), may take
+very different approaches for the implementation of
+different subsystems. They need:
+
+* The API to be abstract and hide implementation details.
+* A specification of the API, ideally with a test suite
+  that ensures compatibility.
+* It would be nice to have an ABI that can be shared
+  across Python implementations.
+
+Alternative APIs
+----------------
+
+There are several projects that implement alternatives to the
+C API, which offer extension users advantanges over programming
+directly with the C API. These APIs are implemented with the
+C API, and in some cases by using CPython internals.
+Examples are
+`HPy <https://hpyproject.org>`__ and
+`pythoncapi-compat <https://pythoncapi-compat.readthedocs.io/en/latest/>`__.
+CPython's DSL for parsing function arguments, the
+`Argument Clinic <https://docs.python.org/3/howto/clinic.html>`__,
```

Comment:
Nitpick: the link here is a human redirect to <https://devguide.python.org/development-tools/clinic/>, so it might be worth just direct linking there instead 🙂

Reply:
Thanks. I updated the link.